### PR TITLE
Add basic test to make sure python runs

### PR DESCRIPTION
**Pull request by rbren**
This comes out of me almost breaking main.py while editing the server.

### CODE REVIEW

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    branches: [ main ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
+      docker:
+        image: docker:19.03.12
+        options: --privileged
+    steps:
+    - uses: actions/checkout@v2
```
Comment (enyst) [inline .github/workflows/test-action.yml:14]: Is there a reason why we should be using v2? It uses node 12 which is deprecated, and there are v3 and v4.

Comment (rbren) [reply .github/workflows/test-action.yml:14]: Hah, the dangers of having an LLM generate code for you 😅

Comment (rbren) [inline .github/workflows/test-action.yml:14]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/test-action.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      docker:
+        image: docker:19.03.12
+        options: --privileged
+    steps:
+    - uses: actions/checkout@v4
+    - name: Run main.py with fake model
+      run: |
+        export SANDBOX_CONTAINER_IMAGE="ubuntu:24.04"
+        python -m pip install -r requirements.txt
+        PYTHONPATH=`pwd` python ./opendevin/main.py -d ./ -t "write a hello world script" --model-name=fake
```
Comment (rbren) [inline .github/workflows/test-action.yml:19]: @xingyaoww would be good to run this a second time with the codeact agent (which I broke in an earlier PR 🙃). Would it be hard to respect the `fake` model and just return a no-op?

Comment (xingyaoww) [reply .github/workflows/test-action.yml:19]: Sounds good, will look at this when i finished #105

**Comment by rbren** [comment]
We have a new `fake` implementation and the linter seems to be doing a good job here. Closing.

**Pull request closed without merging**